src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtlDocumentTransformer.cs

```diff
@@ -344,6 +405,159 @@ private void AddSingleCounterGroup(LazyStringValue docId, BlittableJsonReaderObj
                     _currentRun.AddCounter(docId, prop.Name, value);
             }
         }
+        
+        private void AddTimeSeries(LazyStringValue docId, Dictionary<string, IEnumerable<TimeSeriesStorage.Reader.SingleResult>> timeSeriesReaders)
+        {
+            foreach (var (timeSeriesName, entries) in timeSeriesReaders)
+            {
+                _currentRun.RemoveTimeSeries(docId, timeSeriesName, DateTime.MinValue, DateTime.MaxValue);
```

Comment:
It is temporary.
But I need to figure out how to track which time-series I need to remove 
while they are depended on the document and can be loaded entirely, none or part due to changes in the document, right?

Reply:
@karmeli87 how do we handle "TS tombstones" ?

Reply:
Deleting by tombstones is in progress.
But I referred to a scenario like below:
```
public async Task Test()
{
    const string script = @"
loadToUsers({ Name: this.Name + ' ' + this.LastName });

function loadTimeSeriesOfUsersBehavior(docId, counter)
{
    var user = load(docId);

    if (user.Age < 18)
    {
        return true;
    }
}";
            
    var user = new User {Id = "users/1", Age = 17};
    
    var src = GetDocumentStore(_options);
    var dest = GetDocumentStore();
    
    AddEtl(src, dest, "Users", script);

    using (var session = src.OpenAsyncSession())
    {
        await session.StoreAsync(user);
        session.TimeSeriesFor(user.Id, "Heartrate").Append(DateTime.Now, new []{58.0});
        await session.SaveChangesAsync();
    }

    using (var session = src.OpenAsyncSession())
    {
        session.Advanced.Increment<User, int>(user.Id, x => x.Age, 1);
        await session.SaveChangesAsync();
    }
}
```

Reply:
What commands do we send to a destination db currently after modifying a document like in above test?

Reply:
Do you ask about commands related to time-series?
I just removed all (from DateTime.MinValue to DateTime.MaxValue) and readd the relevant.
But of course, it is not so efficient.
I did the same test for counters to check how we handle it there but as I understand we don't.
Should we?

Reply:
Yes, I meant the commands that ETL sends under the covert to a destination db.

Regarding updates to documents (like `session.Advanced.Increment<User, int>(user.Id, x => x.Age, 1);` in your test) and handling counters then - please see [here](https://issues.hibernatingrhinos.com/issue/RavenDB-11157#focus=streamItem-67-29744.0-0) the general overview - in particular "we send all counters on document update (if not script is defined or the load counter behavior function returns true)"

I don't know if we can do better for Time Series. @karmeli87 ?

Reply:
After thinking about it a bit more I think we can differentiate two cases:

- if a document is sent to the same collection (`users/1` from `Users` is send using `loadToUsers()`) then we preserve the document ID - > `users/1` - then if a document is updated we could skip sending `DELETE users/1` so we must not send _all_ time series

- if a document is sent to different collection then we don't know its ID on destination side so we always sent `DELETE docs where prefix ID is users/1`, then I think we have to send all time series as well

Thoughts?

Reply:
That make sense, but that would be _horrible_ for large systems, no?
It might be better to disable that entirely, if we are sending to a different collection.

Reply:
Oren, we're talking here about the setup that someone _enabled_ it on purpose - by setting up a behavior function like:

```
function loadTimeSeriesOfUsersBehavior(docId, counter)
{
    return true; // not filtering - sent every
}
```
That won't be our default behavior. So I think it's fine - we just need to make people aware of it in the documentation.

Reply:
Okay, then